src/components/common/blocks/gas-price/index.js

```diff
@@ -0,0 +1,213 @@
+
+const MIN_GWEI = 1;
+const MAX_GWEI = 100;
+
+const FASTEST = 'fastest';
+const FAST = 'fast';
+const AVERAGE = 'average';
+const SAFE_LOW = 'safeLow';
+
+const formatTime = totalMinutes => {
+  const hours = Math.floor(totalMinutes / 60);
+  const minutes = totalMinutes % 60;
+  if (hours > 0) {
+    return `${hours} hour${hours > 1 ? 's' : ''}`;
+  } else if (minutes >= 1) {
+    return `${minutes} minute${minutes > 1 ? 's' : ''}`;
+  }
+
+  return `${minutes * 60} seconds`;
+};
+
+class GasPrice extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      gasOption: '',
+      showAdvanced: false,
+      value: 10,
+    };
+  }
+
+  componentDidMount() {
+    fetch(ETH_GAS_STATION_API)
+      .then(res => res.json())
+      .then(
+        result => {
+          const gasData = _.pick(result, [
+            AVERAGE,
+            'avgWait',
+            FAST,
+            'fastWait',
+            FASTEST,
+            'fastestWait',
+            SAFE_LOW,
+            'safeLowWait',
+          ]);
+          const value = gasData[FAST] / 10;
+          this.setState(
+            {
+              gasOption: FAST,
+              showAdvanced: true,
+              value,
+              ...gasData,
+            },
+            () => {
+              this.props.onGasPriceChange(value);
+            }
+          );
+        },
+        error => {
+          console.log(error);
+        }
+      );
+  }
+
+  _onSliderChange = value => {
+    this.setState({ gasOption: '', value }, () => this.props.onGasPriceChange(value));
+  };
+
+  _onSelectOption = gasOption => {
+    const value = this.state[gasOption] / 10;
+    this.setState({ gasOption, value }, () => this.props.onGasPriceChange(value));
+  };
+
+  _toggleAdvanced = () => {
+    this.setState({ showAdvanced: !this.state.showAdvanced });
+  };
+
+  render() {
+    const { gas, theme } = this.props;
+    const { gasOption, showAdvanced, value } = this.state;
+    const transactionFee = web3Utils.fromWei(
+      web3Utils.toWei(String(value * web3Utils.hexToNumber(gas)), 'gwei'),
+      'ether'
+    );
+
+    return (
+      <React.Fragment>
+        <EstimateText>
+          <HeaderText>Estimated Processing Times</HeaderText>
+
+          <span>Select how fast you'd like to send this transaction</span>
+          <TransactionFeeText>
+            Transaction Fee: <ValueText>{transactionFee} ETH</ValueText>
+          </TransactionFeeText>
+        </EstimateText>
+
+        {showAdvanced && (
+          <GasPriceSelect>
+            <GasPriceOption
+              theme={theme}
+              isActive={gasOption === FASTEST}
+              onClick={() => this._onSelectOption(FASTEST)}
+            >
+              <Title>Fastest</Title>
+              <ValueText>
+                <span>{this.state[FASTEST] / 10}</span> GWEI
+              </ValueText>
+              <TimeText>{formatTime(this.state.fastestWait)}</TimeText>
```

Comment:
We missed adding the `theme` prop for `TimeText` here (this and in all other instances of it). This causes the UI to crash.

Reply:
@mandres-digix - Ahh. My bad!
Thanks for pointing it out. I'm out now and will fix it in the next couple of hours 

cc: @mikej-digix